node/malus/src/variants/common.rs

```diff
@@ -236,28 +241,70 @@ where
 								),
 							})
 						}
-						create_validation_response(
-							validation_data,
-							candidate_receipt.descriptor,
-							sender,
-						);
-						None
+						// Create the fake response with probability `p` if the `PoV` is malicious.
+						let distribution = Bernoulli::new(self.percentage / 100.0).expect("Invalid probability! Percentage cannot be < 0 or > 100.");
+						let random_bool = distribution.sample(&mut rand::thread_rng());
```

Comment:
I agree, naming is quite tricky. In this case, `random_bool` is used because it is a random variable (of type boolean) sampled from the distribution. `behave_maliciously` sounds more descriptive at a higher level, however.